### PR TITLE
Fix pre-trained policy action grad leak

### PR DESCRIPTION
**Pull request by AntoineRichard**
# Description

Fixes a latent autograd leak in `PreTrainedPolicyAction.apply_actions`. When environments are stepped outside `torch.inference_mode()`, the low-level policy output can require gradients; copying that output into `self.low_level_actions` makes the persistent action buffer require gradients as well. Warp later rejects that tensor when accessing `__cuda_array_interface__`.

This runs the low-level policy under `torch.no_grad()` and detaches its output before copying it into the action buffer. No new dependencies are required.

@pascal-roth FYI for upstream review.

Fixes: N/A

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

N/A.

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added a changelog fragment under `source/<pkg>/changelog.d/` for every touched package (do **not** edit `CHANGELOG.rst` or bump `extension.toml` — CI handles that)
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

## Verification

- `./isaaclab.sh -f` was run first. All non-LFS hooks passed, but `check-git-lfs-pointers` failed because `git-lfs` is not installed in this environment.
- `SKIP=check-git-lfs-pointers ./isaaclab.sh -f` passed.
- Focused inline check with normal grad mode enabled confirmed the policy output requires grad before the action term runs, while `low_level_actions.requires_grad`, `low_level_actions.grad_fn`, and the downstream action tensor remain detached after `apply_actions()`.
- No automated test was added for this change.

### CODE REVIEW

File: source/isaaclab_tasks/isaaclab_tasks/contrib/navigation/mdp/pre_trained_policy_action.py [hunk 1]
```diff
@@ -96,7 +96,8 @@ def process_actions(self, actions: torch.Tensor):
     def apply_actions(self):
         if self._counter % self.cfg.low_level_decimation == 0:
             low_level_obs = self._low_level_obs_manager.compute_group("ll_policy")
-            self.low_level_actions[:] = self.policy(low_level_obs)
+            with torch.no_grad():
```
Comment (pascal-roth) [inline source/isaaclab_tasks/isaaclab_tasks/contrib/navigation/mdp/pre_trained_policy_action.py]: Are we stepping the simulation at any point that is not already wrapped in torch.no_inference() or torch.no_grad() ? IMO this should not be necessary

**Comment by AntoineRichard** [comment]
@pascal-roth for approval

**Review by pascal-roth — APPROVED** [review]
Amazing thanks
